django_project/changes/tests/test_views.py

```diff
@@ -752,8 +752,368 @@ def test_VersionDelete_no_login(self):
+    """
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def setUp(self):
+        """
+        Setup before each test.
+        We force the locale to en otherwise it will use
+        the locale of the host running the tests and we
+        will get unpredictable results / 404s.
+        """
+
+        self.client = Client()
+        self.client.post(
+            '/set_language/', data={'language': 'en'})
+        logging.disable(logging.CRITICAL)
+        self.project = ProjectF.create(name='testproject')
+        self.version = VersionF.create(
+            project=self.project,
+            name='1.0.1')
+        self.category = CategoryF.create(
+            project=self.project,
+            name='testcategory')
+
+        self.user = None
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def tearDown(self):
+        """
+        Teardown after each test.
+        """
+        self.project.delete()
+        self.version.delete()
+        self.category.delete()
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionListView_with_anonymous_user(self):
+        """
+        Test if anonymous user can view version entry list.
+        """
+        response = self.client.get(reverse('version-list', kwargs={
+            'project_slug': self.project.slug
+        }))
+
+        expected_templates = [
+            'version/list.html', u'changes/version_list.html'
+        ]
+        self.assertEqual(
+            response.status_code, 200)
+        self.assertEqual(
+            response.template_name, expected_templates)
+        self.assertEqual(
+            response.context_data['object_list'][0],
+                         self.version)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionCreateView_with_anonymous_user(self):
+        """
+        Test if anonymous user can create a version entry.
+        """
+        response = self.client.get(reverse('version-create', kwargs={
+            'project_slug': self.project.slug
+        }))
+        self.assertEqual(
+            response.status_code, 302)
+
+    @override_settings(VALID_DOMAIN=['testserver'])
+    def test_VersionUpdateView_with_anonymous_user(self):
+        """
+        Test if anonymous user can update a version entry.
+        """
+        response = self.client.get(reverse('version-create', kwargs={
+            'project_slug': self.project.slug
+        }))
+        self.assertEqual(
+            response.status_code, 302)
+
+    @override_settings(VALID_DOMAIN=['testserver'])
+    def test_VersionDeleteView_with_anonymous_user(self):
+        """
+        Test if anonymous user can delete a version entry.
+        """
+        response = self.client.get(reverse('version-delete', kwargs={
+            'slug': self.version.slug,
+            'project_slug': self.version.project.slug
+        }))
+        self.assertEqual(
+            response.status_code, 302)
```

Comment:
While you tweaking the code layout, you should also use the maximum characters allowed (79 chars). Please update in the test file everywhere where the line can fit.